### PR TITLE
Issue 169

### PR DESCRIPTION
**Pull request by lxning**

### CODE REVIEW

File: gateway/controller.go [hunk 1]
```diff
@@ -178,8 +178,16 @@ func (c *ControllerHTTPClient) SetNamespace(namespace string) {
 }
 
 // FetchAllSchemas fetches all schemas
-func (c *ControllerHTTPClient) FetchAllSchemas() (tables []common.Table, err error) {
-	return c.GetAllSchema(c.namespace)
+func (c *ControllerHTTPClient) FetchAllSchemas() (tables []*common.Table, err error) {
+	schemas, err := c.GetAllSchema(c.namespace)
```
Comment (lucafuji) [inline gateway/controller.go]: here since you are using :=, the err will shadow the err in the return arguments, so anyway you will return a nil err.
I recommend you either using non-named return arguments or pre declare schemas

Comment (lucafuji) [reply gateway/controller.go]: BTW, I think you need to write a test case to catch above non-nil error even it fails

File: gateway/controller_test.go [hunk 2]
```diff
@@ -155,9 +158,9 @@ var _ = ginkgo.Describe("Controller", func() {
 		Ω(err).Should(BeNil())
 
 		c.SetNamespace("ns1")
-		tablesGot, err = c.FetchAllSchemas()
+		tableAddressesGot, err := c.FetchAllSchemas()
 		Ω(err).Should(BeNil())
-		Ω(tablesGot).Should(Equal(tables))
+		Ω(tableAddressesGot).Should(Equal(tableAddresses))
```
Comment (lucafuji) [inline gateway/controller_test.go]: better to add another test case with two table

**Comment by lucafuji** [comment]
BTW give the pull request a better name ...

File: gateway/controller.go (hunk 1 above)
Comment (lxning) [reply gateway/controller.go]: good catch

File: gateway/controller_test.go (hunk 2 above)
Comment (lxning) [reply gateway/controller_test.go]: done
